### PR TITLE
Refactoring propagation context

### PR DESCRIPTION
**Pull request by antonpirker**
Create a class for the `PropagationContext`. Make the class generate the UUIDs lazily.

Fixes https://github.com/getsentry/sentry-python/issues/2827

### CODE REVIEW

File: sentry_sdk/scope.py [hunk 1]
```diff
@@ -431,77 +430,28 @@ def _load_trace_data_from_env(self):
...
-                baggage_header
-            ).dynamic_sampling_context()
-
-        sentry_trace_header = normalized_data.get(SENTRY_TRACE_HEADER_NAME)
-        if sentry_trace_header:
-            sentrytrace_data = extract_sentrytrace_data(sentry_trace_header)
-            if sentrytrace_data is not None:
-                context.update(sentrytrace_data)
-
-        only_baggage_no_sentry_trace = (
```
Comment (sl0thentr0py) [inline sentry_sdk/scope.py:451]: this case is no longer there in the new logic, is it not needed?

**Review by sl0thentr0py — APPROVED** [review]
one comment, ty for refactoring!

File: sentry_sdk/tracing_utils.py [hunk 2]
```diff
@@ -318,6 +319,109 @@ def _format_sql(cursor, sql):
...
+        if baggage_header:
+            propagation_context = PropagationContext()
+            propagation_context.dynamic_sampling_context = Baggage.from_incoming_header(
+                baggage_header
+            ).dynamic_sampling_context()
+
+        sentry_trace_header = normalized_data.get(SENTRY_TRACE_HEADER_NAME)
+        if sentry_trace_header:
+            sentrytrace_data = extract_sentrytrace_data(sentry_trace_header)
+            if sentrytrace_data is not None:
```
Comment (antonpirker) [inline sentry_sdk/tracing_utils.py:378]: It is here @sl0thentr0py

File: sentry_sdk/scope.py (hunk 1 above)
Comment (antonpirker) [reply sentry_sdk/scope.py:451]: see other comment below. it is there.

**Pull request merged**